src/tracers/tracers.cpp

```diff
@@ -24,154 +24,774 @@
+The injection routine requires to first loop on the cells to calculate the size of
+the swarm at the new timestep, and then on the cells again to inject the tracers.
+Due to the stochasticity of the injection, we need a deterministic RNG that will
+return the same random number of cells at both par_for. An attempt of implementing
+such RNG using a cell index based seed is in utils/custom_rng.hpp. Comments welcomed.
+====================================================================================== */
+
+/* ===============================================================================
+EvaluateCriterion: custom function containing the criterion that cells have to ful-
+fill to be elligible for the injection of tracers.
+=============================================================================== */
+
+template <typename View4D>
+KOKKOS_INLINE_FUNCTION bool
+EvaluateCriterion(TracerCriterion crit, View4D prim, const Coordinates_t &coords,
+                  const int k, const int j, const int i, const Real threshold,
+                  const Real mbar_over_kb, const Real jet_radius, const Real jet_offset,
+                  const Real jet_thickness, const int ndim) {
+
+  // Loading coordinates
+  const Real dx = coords.Dxc<1>(k, j, i);
+  const Real dy = coords.Dxc<2>(k, j, i);
+  const Real dz = (ndim == 3) ? coords.Dxc<3>(k, j, i) : 1.0;
+
+  switch (crit) {
+  case TracerCriterion::DensityAbove:
+    return prim(IDN, k, j, i) >= threshold;
+
+  case TracerCriterion::DensityBelow:
+    return prim(IDN, k, j, i) <= threshold;
+
+  case TracerCriterion::TemperatureBelow:
+    return mbar_over_kb * prim(IPR, k, j, i) / prim(IDN, k, j, i) <= threshold;
+
+  case TracerCriterion::TemperatureAbove:
+    return mbar_over_kb * prim(IPR, k, j, i) / prim(IDN, k, j, i) >= threshold;
+
+  case TracerCriterion::Jet: {
+    // Coordinates of the cell center
+    const Real x = coords.Xc<1>(k, j, i);
+    const Real y = coords.Xc<2>(k, j, i);
+    const Real z = (ndim == 3) ? coords.Xc<3>(k, j, i) : 0.0;
+
+    // Cylindrical coordinates
+    const Real r = std::sqrt(x * x + y * y);
+    const Real h = z;
+
+    if (r < jet_radius && std::abs(h) >= jet_offset &&
+        std::abs(h) <= jet_offset + jet_thickness) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  default:
+    return false;
+  }
+}
+
+/* ===============================================================================
+Initialize: reads the input parameters, create the tracer package and create the
+swarm object of each individual populations of tracers.
+=============================================================================== */
+
+// Initializing the tracer packages and swarms
 std::shared_ptr<StateDescriptor> Initialize(ParameterInput *pin) {
-  auto tracer_pkg = std::make_shared<StateDescriptor>("tracers");
+  auto tracers_pkg = std::make_shared<StateDescriptor>("tracers");
+  auto swarm_names = pin->GetVector<std::string>("tracers", "swarm_names");
```

Comment:
And this is likely why all other tests fail.
You're trying to read name before you know tracers are enabled.